File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int res = 0;
        int prev = 0;
        for(int i = 0; i < bank.size(); i++)
        {
            int curr = 0;
            for(int j = 0; j < bank[i].length(); j++)
            {
                if(bank[i][j] == '1')
                {
                    curr++;
                }
            }
            res += prev * curr;
            prev = curr > 0 ? curr : prev;
        }
        return res;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/2244-number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res = 0
        prev = 0
        for floor in bank:
            cur = 0
            for cell in floor:
                if cell == '1':
                    cur += 1
            
            res += cur * prev
            prev = cur if cur > 0 else prev
        return res
